### PR TITLE
Add `<amp-story-auto-analytics>` migration option

### LINKED ISSUE #9045: Add initial `<amp-story-auto-analytics>` migration implementation
## Feature Description

While UX on #7739 is being finalized, we can already implement the actual migration under the hood.

### Implementation Brief

* Add new feature flag for `enableAutoAnalyticsMigration` (or something like that)
* Add a new database migration that sets a flag in the database for everyone with a non-empty tracking ID setting (`web_stories_ga_tracking_id`). We can call the flag `web_stories_using_legacy_analytics` or so. If no tracking ID is set, `web_stories_using_legacy_analytics` will be `false`.
* This new flag should be available via the RES API
* In the editor, in addition to all other settings, retrieve the value of `web_stories_using_legacy_analytics`
* If `web_stories_using_legacy_analytics` is `true`, display explanatory text (exact copy TBD, see #7739 for example) and a "Migrate" button below.
* Clicking on "Migrate" should set `web_stories_using_legacy_analytics` to `false` in the database and ideally show some success snackbar message
* In `Analytics.php`, check value of `web_stories_using_legacy_analytics` and either print a `amp-story-auto-analytics` tag or the regular on.  
  ```html
  <amp-story-auto-analytics gtag-id="UA-123456789-0"></amp-story-auto-analytics>
  ```

### PR DESCRIPTION
**Pull request by swissspidy**
## Context

We want to migrate users to the new `<amp-story-auto-analytics>` component, giving them an opt-in if they used the old configuration so far.

## Summary

This PR adds the initial implementation for the `<amp-story-auto-analytics>` migration, behind a feature flag.

See #7739

## User-facing changes

**New UI option:**

Screenshot 2021-09-16 at 22 54 10

**Demo:**

https://user-images.githubusercontent.com/841956/133685535-e4de6438-c2c8-4dc0-8b70-d50c17afd225.mov

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

**Note:** before testing, ensure you have a tracking ID set.

**Note** To reset the state between tests, visit `wp-admin/options.php` and set `web_stories_using_legacy_analytics` to `1`. This will make the migration button appear again.

This PR can be tested by following these steps:

1. On a published story, verify that you see `<amp-analytics>` in the source code
1. Enable `Auto Analytics` experiment
2. Go to dashboard settings
3. See "Migrate" button under Analytics section
1. On a published story, verify that you now see `<amp-story-auto-analytics>` in the source code

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #9045
See #7739

### CODE REVIEW

File: packages/dashboard/src/app/views/editorSettings/videoCache/index.js [hunk 1]
```diff
@@ -74,10 +74,6 @@ export default function VideoCacheSettings({
...
 
 VideoCacheSettings.propTypes = {
-  isEnabled: PropTypes.bool.isRequired,
+  isEnabled: PropTypes.bool,
   updateSettings: PropTypes.func.isRequired,
 };
-
-VideoCacheSettings.defaultProps = {
-  isEnabled: false,
-};
```
Comment (spacedmonkey) [inline packages/dashboard/src/app/views/editorSettings/videoCache/index.js:83]: Is this change related?

File: packages/dashboard/src/app/views/editorSettings/googleAnalytics/index.js [hunk 2]
```diff
@@ -157,10 +185,50 @@ function GoogleAnalyticsSettings({
...
+        {usingLegacyAnalytics && isAutoAnalyticsMigrationEnabled && (
+          <>
+            <SettingSubheading
+              size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}
+            >
+              <TranslateWithMarkup
+                mapping={{
+                  a: (
+                    <InlineLink
+                      href="https://example.com"
```
Comment (spacedmonkey) [inline packages/dashboard/src/app/views/editorSettings/googleAnalytics/index.js:205]: Let's add a todo here, so we don't forget to update this.

File: packages/dashboard/src/app/views/editorSettings/googleAnalytics/index.js [hunk 3]
```diff
@@ -157,10 +185,50 @@ function GoogleAnalyticsSettings({
...
         <SettingHeading htmlFor="gaTrackingID" as="h3">
           {TEXT.SECTION_HEADING}
+          {usingLegacyAnalytics && isAutoAnalyticsMigrationEnabled && (
+            <StyledNotificationBubble notificationCount={1} isSmall />
+          )}
         </SettingHeading>
         <SettingSubheading size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
           {siteKitDisplayText}
         </SettingSubheading>
+        {usingLegacyAnalytics && isAutoAnalyticsMigrationEnabled && (
```
Comment (spacedmonkey) [inline packages/dashboard/src/app/views/editorSettings/googleAnalytics/index.js:195]: This migration is only a WordPress thing right? This migration is only need for WordPress. Should we call this out and leave a todo for when break up the packages.

File: packages/dashboard/src/app/views/editorSettings/videoCache/index.js (hunk 1 above)
Comment (swissspidy) [reply packages/dashboard/src/app/views/editorSettings/videoCache/index.js:83]: Not related; I was seeing these prop types warning during development and updated it here while I was at it.

`isEnabled` has a default value of `false` in the function signature, so it doesn't neeed `isRequired` nor `defaultProps`.

File: includes/Settings.php [hunk 4]
```diff
@@ -141,6 +148,18 @@ public function register() {
 			]
 		);
 
+		register_setting(
```
Comment (spacedmonkey) [inline includes/Settings.php:151]: We can update the settings test to check this setting is registered.

File: packages/dashboard/src/app/views/editorSettings/googleAnalytics/index.js [hunk 5]
```diff
@@ -223,8 +291,10 @@ function GoogleAnalyticsSettings({
 }
 
 GoogleAnalyticsSettings.propTypes = {
-  handleUpdate: PropTypes.func,
   googleAnalyticsId: PropTypes.string,
+  handleUpdateAnalyticsId: PropTypes.func,
+  usingLegacyAnalytics: PropTypes.bool,
```
Comment (spacedmonkey) [inline packages/dashboard/src/app/views/editorSettings/googleAnalytics/index.js:297]: Should we set default for `usingLegacyAnalytics` to false?

Comment (swissspidy) [reply packages/dashboard/src/app/views/editorSettings/googleAnalytics/index.js:297]: The value is coming from the settings provider and the default is already `false` in `defaultSettingsState`. So this is already covered.

File: packages/dashboard/src/app/views/editorSettings/googleAnalytics/index.js (hunk 3 above)
Comment (swissspidy) [reply packages/dashboard/src/app/views/editorSettings/googleAnalytics/index.js:195]: Yes, it's WordPress-specific. Like a lot of other things in this component, e.g. the Site Kit information. No need for a todo in the code for this though, there will be an in-depth analysis of the dashboard as part of #2918 anyway.

File: packages/dashboard/src/app/views/editorSettings/googleAnalytics/index.js (hunk 5 above)
Comment (spacedmonkey) [reply packages/dashboard/src/app/views/editorSettings/googleAnalytics/index.js:297]: If you had default to false, you wouldn't have had to update lots of the tests.

File: packages/dashboard/src/app/views/editorSettings/googleAnalytics/index.js (hunk 3 above)
Comment (spacedmonkey) [reply packages/dashboard/src/app/views/editorSettings/googleAnalytics/index.js:195]: I think we should be adding todo to new functionality we know is WordPress. Why make more work for ourselves later to find it again?

Comment (swissspidy) [reply packages/dashboard/src/app/views/editorSettings/googleAnalytics/index.js:195]: Such a todo doesn't need to be in the code, it can be noted externally in a doc.

**Review by spacedmonkey — APPROVED** [review]
One nitpic, but otherwise,we are looking good!

**Review by spacedmonkey — APPROVED** [review]
